sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosQueryRequestOptions.java

```diff
@@ -71,6 +75,31 @@ CosmosQueryRequestOptions setPartitionKeyRangeIdInternal(String partitionKeyRang
         return this;
     }
 
+    /**
+     * Gets the consistency level required for the request.
+     *
+     * @return the consistency level.
+     */
+
+    public ConsistencyLevel getConsistencyLevel() {
+        return consistencyLevel;
+    }
+
+    /**
+     * Sets the consistency level required for the request. The effective consistency level
+     * can only be reduce for read/query requests. So when the Account's default consistency level
+     * is for example Session you can specify on a request-by-request level for individual requests
+     * that Eventual consistency is sufficient - which could reduce the latency and RU charges for this
+     * request but will not guarantee session consistency (read-your-own-write) anymore
+     *
+     * @param consistencyLevel the consistency level.
+     * @return the CosmosItemRequestOptions.
+     */
+    public CosmosQueryRequestOptions setConsistencyLevel(ConsistencyLevel consistencyLevel) {
```

Comment:
Are we now supporting request level consistency now?

Reply:
We have always been supporting it in .Net - In Java V4 I made it public for read/query where it most commonly used my customers